contracts/Transfer.sol

```diff
@@ -290,4 +294,27 @@ contract Transfer is FraudProofHelpers {
             true
         );
     }
+
+    function ApplyTransferTx(
+        Types.AccountMerkleProof memory _merkle_proof,
+        bytes memory txs,
+        uint256 i
+    ) public view returns (bytes memory updatedAccount, bytes32 newRoot) {
+        Types.UserAccount memory stateLeaf = _merkle_proof.accountIP.account;
+        uint256 stateIndex = _merkle_proof.accountIP.pathToAccount;
+        if (stateIndex == txs.transfer_fromIndexOf(i)) {
```

Comment:
As @vaibhavchellani noted [here](https://github.com/thehubbleproject/contracts/issues/94) this fix should be considered more like a workaround.

Since we already have stateIndex of sender and receiver in transaction data we should use directly from there.

Using an explicit index for state tree:

* introduces redundant data
* leads to confusion (as can be seen up there we don't check the case where `stateIndex` does not belong to neigther receiver nor sender)

Reply:
So since we are not relying on pathToAccount, can we do two ApplyTx functions? i.e. ApplyTxFrom, ApplyTxTo

Reply:
I'm asking @vaibhavchellani